torch/_functorch/_aot_autograd/graph_capture.py

```diff
@@ -74,12 +98,13 @@ def _detach_and_copy_item_memo(t):
 
 
 def aot_dispatch_base_graph(
-    flat_fn,
-    flat_args: list[Tensor],
+    flat_fn: TraceFn,
+    flat_args: list[FxValue],
+    flat_args_descs: list[AOTInput],
```

Comment:
So today, fw_metadata : ViewAndMutationMeta contains things like InputAlias information and other things assigned to each input. Is there a reason you chose to thread descriptors through over putting them inside the metadata? I can see it being easier to reason about by explicitly threading it through, if that's the main reason

Reply:
In fact, the problem is specifically that it is completely impossible to interpret the metadata as a non expert user. I do agree that in principle everything I have here could be replicated purely from the metadata data structures preexisting in AOTAutograd. Good luck figuring it out!